src/python_bluesky_taskgraph/core/task_graph.py

```diff
@@ -1,13 +1,29 @@
-from typing import Dict, List, Set
+from dataclasses import dataclass
+from typing import Callable, Dict, List, Set, Union
 
 from python_bluesky_taskgraph.core.task import BlueskyTask
+from python_bluesky_taskgraph.tasks.stub_tasks import CloseRunTask, OpenRunTask
+
+
+@dataclass
+class TaskTuple:
```

Comment:
Might think about renaming this, only because `Tuple` is an overloaded word (`NamedTuple` et al.) and I got confused

```suggestion
class PreparedTask:
```